drivers/video/ov7670.c

```diff
@@ -341,6 +344,27 @@ static const struct ov7670_reg ov7670_init_regtbl[] = {
 	{0xb8, 0x0a},
 };
 
+static int ov7670_update_reg(const struct device *dev, uint8_t reg, uint32_t mask, uint32_t new_val)
```

Comment:
i2c_reg_update_byte_dt ? :)

Reply:
Updated. I also fixed TSLB[3]. I tested RGB and Grayscale from YUV with these fixes, and confirm things work fine.

Reply:
Sorry if I wasn't clear, I meant you can just drop this function and use the built-in one from the I2C driver instead?

Reply:
Wow I didn't know there's one. Okay.

Reply:
Yeah should have spotted the pattern earlier when I first suggested you factor some of the code, sorry. That's now a lot simpler than originally eh? :)

Reply:
No worries, yes, it's much simpler now.